src/lib/Microsoft.Fx.Portability/Proxy/ProxyAuthenticationHandler.cs

```diff
@@ -41,7 +43,10 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
             {
                 try
                 {
-                    return await base.SendAsync(request, cancellationToken);
+                    HttpResponseMessage result = await base.SendAsync(request, cancellationToken);
+                    this.LastStatusCode = result.StatusCode;
```

Comment:
Storing the last status code in a field leaves it open to races between multiple requests. Instead of this pattern, we could throw our own exception that is handled similarly in ProxyAuthenticationRequied().

```suggestion
                    if (this.LastStatusCode == HttpStatusCode.ProxyAuthenticationRequired)
                    {
                        throw SomePrivateExceptionWeHandleBelow();
                    }
```

Reply:
Let me know if you're not following my suggestion

Reply:
Yeah, i got it.